docs/CustomMiddlewareChain.md

```diff
@@ -0,0 +1,148 @@
+# Using Custom Middleware Chain
+
+## Middleware
+
+As name suggests it comes in middle of something and that is request and response cycle. It has access to context object which contains request, response objects and any other options that are specific to middleware. Also it has an access to execute next middleware in the chain.
+
+## Step by step procedure
+
+### Implement Middlewares
+
+Create own set of middlewares by implementing [Middleware](../src/IMiddleware.ts) interface. Here two middlewares are created one for handling Logging and another for handling http request and response.
```

Comment:
"Create own set of middlewares by implementing Middleware interface. Here two middlewares are created one for handling Logging and another for handling http request and response."

"Create a custom middleware pipeline by implementing the IMiddleware interface. The following examples demonstrate how to create a custom logging middleware and how to create a custom http request a response handler."